src/Layout/redist/targets/GenerateMSIs.targets

```diff
@@ -233,12 +242,45 @@
     </ItemGroup>
   </Target>
 
-  <Target Name="GenerateWorkloadManifestsWxs">
+  <Target Name="UnzipBuiltinWorkloadManifestMsiNuPkg">
+    <PropertyGroup>
+      <BuiltinManifestMsiPackageSourceDirectory>$(ArtifactsShippingPackagesDir)</BuiltinManifestMsiPackageSourceDirectory>
+      <BuiltinManifestMsiPackageSourceDirectory Condition="'$(DotNetBuild)' != 'true' and '$(BuildWorkloads)' != 'true' and '$(OfficialBuildId)' != ''">$(ArtifactsDir)downloaded-manifest-msi-packages\PackageArtifacts</BuiltinManifestMsiPackageSourceDirectory>
+      <BuiltinManifestMsiPackageSourceDirectory Condition="'$(DotNetBuild)' == 'true' and '$(DotNetBuildPass)' == '3'">$(SourceBuiltShippingPackagesDir)</BuiltinManifestMsiPackageSourceDirectory>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <_builtinManifestMsiPackages Include="$(BuiltinManifestMsiPackageSourceDirectory)/%(BuiltinManifests.Identity).Manifest-*.Msi.$(MsiArchitectureForWorkloadManifests)*.nupkg" />
+      <BuiltinManifestMsiPackages Include="@(_builtinManifestMsiPackages)" UnzipDestinationFolder="$(IntermediateOutputPath)\extracted-manifests\%(Filename)" />
+    </ItemGroup>
+
+    <PropertyGroup>
+      <!-- for local SDK builds we allow skipping the workload manifest msi generation to save build time, the sdk bundle won't contain the manifest msis in that case. -->
+      <AllowEmptyBuiltinWorkloadManifestMsi Condition="'$(DotNetBuild)' != 'true' and '$(BuildWorkloads)' != 'true' and '$(OfficialBuildId)' == ''">true</AllowEmptyBuiltinWorkloadManifestMsi>
+    </PropertyGroup>
+
+    <Warning Text="Builtin manifest msi packages were skipped, the sdk bundle won't contain the builtin manifests and workload installs will be broken."
+             Code="MISSING_MANIFEST_MSI"
+             Condition="@(BuiltinManifestMsiPackages -> Count()) == 0 and '$(AllowEmptyBuiltinWorkloadManifestMsi)' == 'true'" />
+
+    <Error Text="Could not find expected builtin manifest msi packages in $(BuiltinManifestMsiPackageSourceDirectory)"
+           Condition="@(BuiltinManifestMsiPackages -> Count()) == 0 and '$(AllowEmptyBuiltinWorkloadManifestMsi)' != 'true'" />
+
+    <Unzip SourceFiles="@(BuiltinManifestMsiPackages)"
+           DestinationFolder="%(BuiltinManifestMsiPackages.UnzipDestinationFolder)"
+           Condition="@(BuiltinManifestMsiPackages -> Count()) != 0" />
```

Comment:
FWIW, by using PackageDownload (my feedback on using indirection), the Unzip will already be taken care of.